imagehash/__init__.py

```diff
@@ -116,10 +116,10 @@ def average_hash(image, hash_size=8):
 def phash(image, hash_size=32):
 	image = image.convert("L").resize((hash_size, hash_size), Image.ANTIALIAS)
 	pixels = numpy.array(image.getdata(), dtype=numpy.float).reshape((hash_size, hash_size))
-	dct = scipy.fftpack.dct(pixels)
-	dctlowfreq = dct[:8, 1:9]
-	avg = dctlowfreq.mean()
-	diff = dctlowfreq > avg
+	dct = scipy.fftpack.dct(scipy.fftpack.dct(pixels, axis=0), axis=1)
+	dctlowfreq = dct[:8, :8]
```

Comment:
I tried to follow http://www.hackerfactor.com/blog/index.php?/archives/432-Looks-Like-It.html
In step 6, "Further reduce the DCT.", they make bits out of it. The reason the image http://lh6.googleusercontent.com/-kxozFTZ1CSE/Td8I2HJeLoI/AAAAAAAAB4A/IeFNcx4lRQQ/s800/Alyson_Hannigan_200512-phash-dct.png looks grayscale is the anti-aliasing of PNG in the image viewer -- it is actually binary.

Also keep in mind that imshow rotates the matrix.

One row is excluded because those are the constants (total power) in the transform, and the total intensity should not contribute any information. That is at least how the algorithm I used stores the information. 

That said, it is possible that what I programmed deviates algorithmically from what the reference URL did and thus produces different output.